src/core/wallet/hw-wallet/ledger/ledger-wallet.ts

```diff
@@ -0,0 +1,83 @@
+import { IWallet } from '../../types';
+import { Blockchain, IBlockchainTransaction } from '../../../blockchain/types';
+import { IAccountState } from '../../../../redux/wallets/state';
+import { HWModel, HWConnection } from '../types';
+import { AppFactory } from './apps-factory';
+import { TransportFactory } from './transport-factory';
+import { delay } from '../../../utils/time';
+
+export class LedgerWallet implements IWallet {
+    private deviceId: string;
+    private deviceModel: HWModel;
+    private connectionType: HWConnection;
+
+    constructor(deviceModel: HWModel, connectionType: HWConnection, deviceId: string) {
+        this.deviceId = deviceId;
+        this.deviceModel = deviceModel;
+        this.connectionType = connectionType;
+    }
+
+    public onAppOpened(blockchain: Blockchain): Promise<void> {
+        return new Promise(async resolve => {
+            let opened = false;
+            while (opened === false) {
+                try {
+                    const transport = await this.getTransport();
+                    const app = await AppFactory.get(blockchain, transport);
+                    if (await app.getInfo()) {
+                        opened = true;
+                    }
+                } catch {
+                    // dont handle error - keep trying until user opens the app
+                }
+                await delay(1000);
+            }
+            resolve();
+        });
+    }
+
+    public async getAccounts(
+        blockchain: Blockchain,
+        index: number,
+        indexTo?: number
+    ): Promise<IAccountState[]> {
+        indexTo = indexTo || index;
+        const accounts = [];
+
+        try {
+            try {
```

Comment:
this try is not needed for 2 reasons
1. you are already in a try catch
2. onAppOpened never returns a Promise.reject